WeatherMonitoringService.Tests/JsonWeatherParserTest.cs

```diff
@@ -0,0 +1,42 @@
+using FluentValidation.Results;
+using Moq;
+using WeatherMonitoringService.Models;
+using WeatherMonitoringService.Parsers;
+using WeatherMonitoringService.Parsers.Exceptions;
+using ValidationResult = FluentValidation.Results.ValidationResult;
+
+namespace WeatherMonitoringService.Tests;
+
+public class JsonWeatherParserTest
+{
+    
+    [Fact]
+    public void ParseWeatherInput_ShouldReturnsWeatherData()
+    {
+        const string input = "{\"Location\":\"location\",\"Temperature\":25,\"Humidity\":60}";
+        
+        var validatorMock = new Mock<WeatherDataValidator> { CallBase = true };
+
+        var parser = new JsonWeatherDataParser(validatorMock.Object);
+
+        var weatherData = parser.ParseWeatherInput(input);
+
+        Assert.Equal("location", weatherData.Location);
+        Assert.Equal(25, weatherData.Temperature);
+        Assert.Equal(60, weatherData.Humidity);
+    }
```

Comment:
Please follow this convention in all tests
```suggestion
    [Fact]
    public void ParseWeatherInput_ShouldReturnsWeatherData()
    {
        // Arrange
        const string locationName = "location";
        const int temperature = 25;
        const int humidity = 60;
        var expected = new WeatherData { Location = locationName, Temperature = temperature, Humidity = humidity };
        
        var input = $$"""
                      {
                          "Location":"{{locationName}}",
                          "Temperature":{{temperature}},
                          "Humidity":{{humidity}}
                      }
                      """;
        
        // Act
        var weatherData = _parser.ParseWeatherInput(input);

        // Assert
        Assert.Equal(expected.Location, weatherData.Location);
    }
```